src/auto-evo/mutation_strategy/UpgradeOrganelle.cs

```diff
@@ -7,12 +7,19 @@
 public class UpgradeOrganelle : IMutationStrategy<MicrobeSpecies>
 {
     private readonly FrozenSet<OrganelleDefinition> allOrganelles;
-    private readonly IComponentSpecificUpgrades upgrade;
+    private readonly IComponentSpecificUpgrades? customUpgrade;
+    private readonly string? upgradeName;
 
-    public UpgradeOrganelle(Func<OrganelleDefinition, bool> criteria, IComponentSpecificUpgrades upgrade)
+    public UpgradeOrganelle(Func<OrganelleDefinition, bool> criteria, IComponentSpecificUpgrades customUpgrade)
     {
         allOrganelles = SimulationParameters.Instance.GetAllOrganelles().Where(criteria).ToFrozenSet();
-        this.upgrade = upgrade;
+        this.customUpgrade = customUpgrade;
+    }
+
+    public UpgradeOrganelle(Func<OrganelleDefinition, bool> criteria, string upgradeName)
+    {
+        allOrganelles = SimulationParameters.Instance.GetAllOrganelles().Where(criteria).ToFrozenSet();
+        this.upgradeName = upgradeName;
```

Comment:
Shouldn't this verify the upgrade name actually exists in the organelle? And throw an exception as that's a really big game config error.